luminex/src/org/labkey/luminex/query/ExclusionUIDisplayColumn.java

```diff
@@ -90,37 +102,36 @@ public void renderGridCellContents(RenderContext ctx, Writer out) throws IOExcep
 
         String id = "__changeExclusions__" + wellID;
 
-        // add onclick handler to call the well exclusion window creation function
         boolean canEdit = _container.hasPermission(_user, UpdatePermission.class);
-        if (canEdit)
-        {
-            out.write("<a onclick=\"openExclusionsWellWindow(" + _protocolId + ", " + runId + ", " + dataId + ", '" + wellID + "', "
-                + (description == null ? null : "'" + description + "'") + ", '" + type + "');\">");
-        }
-
         Boolean excluded = (Boolean)ctx.get(getColumnInfo().getFieldKey());
-        if (excluded.booleanValue())
+
+        HtmlString img = excluded.booleanValue() ?
+            getImgTag("excluded.png", exclusionComment, id, canEdit) :
+            getImgTag("included.png", "Click to add a well or replicate group exclusion", id, canEdit);
+
+        if (canEdit)
         {
-            out.write("<img src=\"" + AppProps.getInstance().getContextPath() + "/luminex/exclusion/excluded.png\" height=\"16\" width=\"16\" id=\""+id+"\"");
-            if (canEdit)
-            {
-                String tooltip = PageFlowUtil.filter(exclusionComment);
-                out.write("title=\"" + tooltip + "\" alt=\"" + tooltip + "\" ");
-            }
-            out.write(" />");
+            // add onclick handler to call the well exclusion window creation function
+            String onClick = "openExclusionsWellWindow(" + _protocolId + ", " + runId + ", " + dataId + ", '" + wellID +
+                "', " + (description == null ? null : "'" + description + "'") + ", '" + type + "');";

```

Comment:
Use PageflowUtil.jsStrong() instead of assuming that the description is safe

Reply:
Well, it was assumed safe before my change... but okay. :)